gammapy/background/ring.py

```diff
@@ -2,22 +2,236 @@
 """Ring background estimation.
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
+from itertools import product
 import numpy as np
-from astropy.convolution import Ring2DKernel
+from astropy.convolution import Ring2DKernel, Tophat2DKernel
+import astropy.units as u
 from ..image import SkyImageList, SkyImage
+from ..image.utils import scale_cube_fft
 
 
 __all__ = [
+    'AdaptiveRingBackgroundEstimator',
     'RingBackgroundEstimator',
     'ring_r_out',
     'ring_area_factor',
     'ring_alpha',
 ]
 
 
-class RingBackgroundEstimator(object):
+class AdaptiveRingBackgroundEstimator(object):
+    """
+    Adaptive ring background algorithm.
+
+    This alogrithm extends the standard `RingBackground` method by adpating the
+    size of the ring to achieve a minimum on / off exposure ratio (alpha) in regions
+    where the area to estimate the background from is limited.
+
+    Here is an illustration of the method:
+
+    Parameters
+    ----------
+    r_in : `~astropy.units.Quantity`
+        Inner radius of the ring.
+    r_out_max : `~astropy.units.Quantity`
+        Maximal outer radius of the ring.
+    width : `~astropy.units.Quantity`
+        Width of the ring.
+    stepsize : `~astropy.units.Quantity` (0.02 deg)
+        Stepsize used for increasing the radius.
+    threshold : float (0.1)
+        Threshold above which the adaptive ring takes action.
+    theta : `~astropy.units.Quantity` (0.22 deg)
+        Integration radius used for alpha computation.
+    method : {'const. width', 'const. r_in'}
```

Comment:
Maybe use something that's a bit easier to type correctly (no `. `) and can be used e.g. in filenames more easily, e.g. `fix_width` and `fix_r_in` ?